examples/plugin/gimbal_controller/GimbalControllerPlugin.cc

```diff
@@ -0,0 +1,450 @@
+  ignition::msgs::IMU imuMsg = _msg;
+  this->imuMsgValid = true;
+}
+
+/////////////////////////////////////////////////////////////////////////////
+void GimbalControllerPlugin::Configure(const Entity &_entity,
+    const std::shared_ptr<const sdf::Element> &_sdf,
+    EntityComponentManager &_ecm,
+    EventManager &/*_eventMgr*/)
+{
+	this->dataPtr->model = ignition::gazebo::Model(_entity);
+
+	std::string yawJointName = "cgo3_vertical_arm_joint";
+	this->dataPtr->yawJoint = this->dataPtr->model.JointByName(_ecm, "yawJointName");
+
+	if(_sdf->HasElement("joint_yaw"))
+	{
+		// Add names to map
+		yawJointName = _sdf->Get<std::string>("joint_yaw");
+
+		if(this->dataPtr->model.JointByName(_ecm, "yawJointName"))
+		{
+			this->dataPtr->yawJoint = this->dataPtr->model.JointByName(_ecm, "yawJointName");
+		}
+		else
+		{
+			ignwarn << "joint_yaw ["<< yawJointName <<"] does not exist?\n"<< std::endl;
+		}
+	}
+
+	if(!this->dataPtr->yawJoint)
+	{
+		ignerr << "GimbalControllerPlugin::Configure ERROR! Can't get yaw joint. " <<yawJointName <<"' " <<std::endl;
+	}
+
+	// for roll joint
+	std::string rollJointName = "cgo3_horizontal_arm_joint";
+	this->dataPtr->rollJoint = this->dataPtr->model.JointByName(_ecm, "rollJointName");
+
+	if(_sdf->HasElement("joint_roll"))
+	{
+		// Add names to map
+		rollJointName = _sdf->Get<std::string>("joint_roll");
+
+		if(this->dataPtr->model.JointByName(_ecm, "rollJointName"))
+		{
+			this->dataPtr->rollJoint = this->dataPtr->model.JointByName(_ecm, "rollJointName");
+		}
+
+		else
+		{
+			ignwarn << "joint_roll ["<< rollJointName <<"] does not exist?\n"<< std::endl;
+		}
+	}
+
+	if(!this->dataPtr->rollJoint)
+	{
+		ignerr << "GimbalControllerPlugin::Configure ERROR! Can't get roll joint. " <<rollJointName <<"' " <<std::endl;
+	}
+
+	// for pitch joint
+
+	std::string pitchJointName = "cgo3_camera_joint";
+	this->dataPtr->pitchJoint = this->dataPtr->model.JointByName(_ecm, "pitchJointName");
+
+	if(_sdf->HasElement("joint_pitch"))
+	{
+		// Add names to map
+		pitchJointName = _sdf->Get<std::string>("joint_pitch");
+		if(this->dataPtr->model.JointByName(_ecm, "pitchJointName"))
+		{
+			this->dataPtr->pitchJoint = this->dataPtr->model.JointByName(_ecm, "picthJointName");
+		}
+
+		else
+		{
+			ignwarn << "joint_pitch ["<< pitchJointName <<"] does not exist?\n"<< std::endl;
+		}
+	}
+
+	if(!this->dataPtr->pitchJoint)
+	{
+		ignerr << "GimbalControllerPlugin::Configure ERROR! Can't get pitch joint. " <<pitchJointName <<"' " <<std::endl;
+	}
+	// get imu sensor
+	this->dataPtr->imuName = _sdf->Get("imuName", static_cast<std::string>("imu_sensor")).first;
+}
+
+///////////////////////////////////////////////////////////////////////////
+void GimbalControllerPluginPrivate::Init(const EntityComponentManager &_ecm)
+{    
+  // receive pitch command via ignition transport 
+	std::string pitchTopic = std::string("~/") +  this->model.Name(_ecm) + "/gimbal_pitch_cmd";
```

Comment:
this symbol `~/` looks like a ROS like topic, please fix this

Reply:
you want me to remove this?